### PR TITLE
fix: pipeline deployment

### PR DESCRIPTION
**Pull request by laurentiu-ghita**
- re add primaryOutputDirectory since without it build step was failing

- add missing policies for app build step

### CODE REVIEW

**Review by completenovice — DISMISSED** [review]

File: release/infrastructure/lib/cicd/deployment-pipeline.ts [hunk 1]
```diff
@@ -142,6 +149,14 @@ export class DeploymentPipeline extends Construct {
         pipeline: pipelines.CodePipeline,
         { codeBuildImage, vpc, subnetSelection }: CodeBuildVpcSettings
     ): pipelines.FileSet {
+        const role = new iam.Role(this, 'build-project-role', {
+            assumedBy: new iam.ServicePrincipal('codebuild.amazonaws.com')
+        });
```
Comment (completenovice) [inline release/infrastructure/lib/cicd/deployment-pipeline.ts:154]: What's this being used for?

Comment (laurentiu-ghita) [reply release/infrastructure/lib/cicd/deployment-pipeline.ts:154]: I updated the PR.
it's used down below in the CodeBuildStep

File: release/infrastructure/lib/cicd/deployment-pipeline.ts [hunk 2]
```diff
@@ -155,6 +170,7 @@ export class DeploymentPipeline extends Construct {
             commands: [], // leave empty to use the buildspec
             vpc,
             subnetSelection,
+            role,
             rolePolicyStatements: [...CiCdUtilityHelper.getArtifactRegistriesPolicyStatementsHelper()]
```
Comment (completenovice) [inline release/infrastructure/lib/cicd/deployment-pipeline.ts:174]: Minor: We could probably add these to the role directly, instead of mixing them here

**Review by completenovice — APPROVED** [review]

File: release/infrastructure/lib/cicd/deployment-pipeline.ts (hunk 2 above)
Comment (laurentiu-ghita) [reply release/infrastructure/lib/cicd/deployment-pipeline.ts]: fixed

**Pull request merged**